From 5b9c0449c0d8ac2d42a66ea81ff2c4b8dca50a33 Mon Sep 17 00:00:00 2001
From: Patrice Chotard <patrice.chotard@foss.st.com>
Date: Fri, 17 Nov 2023 09:53:22 +0100
Subject: [PATCH 0658/1141] arm64: dts: st: enable hdmi audio on stm32mp257f-dk

Add HDMI audio support through ADV7535 HDMI transceiver on
STM32MP257F-D board.
The sound card and i2s2 nodes are disabled by default.
If the HDMI MB1752 board is detected, these nodes are enabled
in uboot at runtime.

Signed-off-by: Olivier BIDEAU <olivier.bideau@st.com>
Signed-off-by: Patrice Chotard <patrice.chotard@foss.st.com>
Change-Id: I168d7b920fab6d0240be6ad6dec347ab6723c708
---
 arch/arm64/boot/dts/st/stm32mp257f-dk.dts | 33 +++++++++++++++++++++++
 1 file changed, 33 insertions(+)

--- a/arch/arm64/boot/dts/st/stm32mp257f-dk.dts
+++ b/arch/arm64/boot/dts/st/stm32mp257f-dk.dts
@@ -117,6 +117,17 @@
 		status = "okay";
 	};
 
+	sound {
+		compatible = "audio-graph-card";
+		label = "STM32MP25-DK";
+		routing =
+			"Playback" , "MCLK",
+			"Capture" , "MCLK",
+			"MICL" , "Mic Bias";
+		dais = <&sai1a_port &sai1b_port &i2s2_port>;
+		status = "okay";
+	};
+
 	wifi_pwrseq: wifi-pwrseq {
 		compatible = "mmc-pwrseq-simple";
 		reset-gpios = <&gpiog 8 GPIO_ACTIVE_LOW>;
@@ -222,6 +233,13 @@
 					remote-endpoint = <&hdmi_connector_in>;
 				};
 			};
+
+			port@2 {
+				reg = <2>;
+				adv7535_tx_endpoint: endpoint {
+					remote-endpoint = <&i2s2_endpoint>;
+				};
+			};
 		};
 	};
 
@@ -281,6 +299,21 @@
 	/delete-property/dma-names;
 };
 
+&i2s2 {
+	pinctrl-names = "default", "sleep";
+	pinctrl-0 = <&i2s2_pins_b>;
+	pinctrl-1 = <&i2s2_sleep_pins_b>;
+	status = "okay";
+
+	i2s2_port: port {
+		i2s2_endpoint: endpoint {
+			remote-endpoint = <&adv7535_tx_endpoint>;
+			format = "i2s";
+			mclk-fs = <256>;
+		};
+	};
+};
+
 &ltdc {
 	status = "okay";
 
